pjsip/include/pjsip-ua/sip_inv.h

```diff
@@ -459,15 +459,19 @@ struct pjsip_inv_session
     pj_atomic_t		*ref_cnt;		    /**< Reference counter. */
     pj_bool_t            updated_sdp_answer;        /**< SDP answer just been
 							 updated?	    */
+    pj_bool_t            create_multipart;          /**< Create multipart
+                                                         bodies.  Should be
+                                                         used for testing
+                                                         only.              */
```

Comment:
Not sure about this setting because:
- Potential behavioral change affecting app or upper level API.  For example, currently pjsua seems to always assume that Invite module will always create non-multipart body, see how this [pjsua_process_msg_data()](https://github.com/pjsip/pjproject/blob/master/pjsip/src/pjsua-lib/pjsua_core.c#L2994) implementation creates multipart body, it does not check whether the existing body, which is generated by Invite module, is a multipart (as specified [here](https://datatracker.ietf.org/doc/html/rfc5621#section-4.3), nested multipart is not recommended). Some apps may assume the same.
- Even if the ```pjsip_inv_session``` structure is public (as opposed to transparent), as far as I know, modifying/writing the field directly should be avoided, it should be done via a function (or an option/flag), which does not seem to be provided here.

Also, as currently this setting seems to be used by a test app only, so perhaps better to handle multipart body generation in the test app if possible?

Reply:
Hmmm.   That flag really was there for internal testing only as the comment states.  I wanted the existing unit tests in inv_offer_answer_test to be able to exercise the  multipart logic in sip_inv.  Having the test do the multipart handling would defeat the purpose.  I can just remove the extra tests and the field if you prefer.

Reply:
I mean the app should generate the multipart body (instead of handling the whole multipart things) and leave sip_inv to generate non-multipart body as is (so the new field `create_multipart` can be avoided), something like this: [2920_diff.txt](https://github.com/pjsip/pjproject/files/7724906/2920_diff.txt).
.

Reply:
Ah, gotcha.  Let me try that and see if it still exercises the code paths.

Reply:
OK, I removed the create_multipart option and applied the diff to inv_offer_answer_test.  Seems to work fine.  Anything else I need to do?